WzLib/WzSerializer.cs

```diff
@@ -188,6 +188,197 @@ protected void WritePropertyToXML(TextWriter tw, string depth, WzImageProperty p
+                        $"\"basedata\": {Convert.ToBase64String(pngbytes)}\",");
+                }
+                else
+                    tw.Write($"{{" +
+                        $"\"width\": {property.PngProperty.Width}, " +
+                        $"\"height\": {property.PngProperty.Height},");
+                if (property.WzProperties.Count() > 0)
+                {
+                    var last = property.WzProperties.Last();
+                    foreach (WzImageProperty p in property.WzProperties)
+                    {
+                        WritePropertyToJson(tw, p);
+                        if (!p.Equals(last))
+                        {
+                            tw.Write(",");
+                        }
+                    }
+                }
+                tw.Write("}");
+
+            }
+            else if (prop is WzIntProperty)
+            {
+                WzIntProperty property = (WzIntProperty)prop;
+                tw.Write($"\"{XmlUtil.SanitizeText(property.Name)}\": {property.Value}");
+            }
+            else if (prop is WzDoubleProperty)
+            {
+                WzDoubleProperty property = (WzDoubleProperty)prop;
+                tw.Write($"\"{XmlUtil.SanitizeText(property.Name)}\": {property.Value}");
+            }
+            else if (prop is WzNullProperty)
+            {
+                WzNullProperty property = (WzNullProperty)prop;
+                tw.Write($"\"{XmlUtil.SanitizeText(property.Name)}\": null");
+            }
+            else if (prop is WzSoundProperty)
+            {
+                WzSoundProperty property = (WzSoundProperty)prop;
+                if (!isArray)
+                {
+                    tw.Write($"\"{XmlUtil.SanitizeText(property.Name)}\":");
+                }
+                if (ExportBase64Data)
+                    tw.Write($"{{" +
+                        $"\"length\":\"{property.Length}\", " +
+                        $"\"basehead\": \"{Convert.ToBase64String(property.Header)}\"" +
+                        $"\"basedata\": \"{Convert.ToBase64String(property.GetBytes(false))}\"" +
+                        $"}}");
+                else
+                    tw.Write("{}");
+            }
+            else if (prop is WzStringProperty)
+            {
+                WzStringProperty property = (WzStringProperty)prop;
+                tw.Write($"\"{XmlUtil.SanitizeText(property.Name)}\": {JsonConvert.ToString(property.Value)}");
+            }
+            else if (prop is WzSubProperty)
+            {
+                WzSubProperty property = (WzSubProperty)prop;
+                if (!isArray)
+                {
+                    tw.Write($"\"{XmlUtil.SanitizeText(property.Name)}\":");
+                }
+                // This has the same problem as the convex property
+                bool propertyIsArray = property.WzProperties.TrueForAll(x => { int num; return int.TryParse(x.Name, out num); });
```

Comment:
We treat arrays differently if all of the keys are numeric. The object underneath will get wrapped in a structure with an `index` and `item` field.